extensions/b2g/viewer.js

```diff
@@ -0,0 +1,366 @@
+  },
+
+  zoomIn: function pdfViewZoomIn(ticks) {
+    var newScale = this.pdfViewer.currentScale;
+    do {
+      newScale = (newScale * DEFAULT_SCALE_DELTA).toFixed(2);
+      newScale = Math.ceil(newScale * 10) / 10;
+      newScale = Math.min(MAX_SCALE, newScale);
+    } while (--ticks && newScale < MAX_SCALE);
+    this.setScale(newScale, true);
+  },
+
+  zoomOut: function pdfViewZoomOut(ticks) {
+    var newScale = this.pdfViewer.currentScale;
+    do {
+      newScale = (newScale / DEFAULT_SCALE_DELTA).toFixed(2);
+      newScale = Math.floor(newScale * 10) / 10;
+      newScale = Math.max(MIN_SCALE, newScale);
+    } while (--ticks && newScale > MIN_SCALE);
+    this.setScale(newScale, true);
+  },
+
+  setScale: function (value, resetAutoSettings) {
+    this.updateScaleControls = !!resetAutoSettings;
+    this.pdfViewer.currentScaleValue = value;
+    this.updateScaleControls = true;
+  },
+
+  initUI: function pdfViewInitUI() {
+    var linkService = new PDFJS.PDFLinkService();
+    this.pdfLinkService = linkService;
+
+    var container = document.getElementById('viewerContainer');
+    var pdfViewer = new PDFJS.PDFViewer({
+      container: container,
+      linkService: linkService
+    });
+    this.pdfViewer = pdfViewer;
+    linkService.setViewer(pdfViewer);
+
+    this.pdfHistory = new PDFJS.PDFHistory({
+      linkService: linkService
+    });
+    linkService.setHistory(this.pdfHistory);
+
+    document.getElementById('previous').addEventListener('click', function() {
+      PDFViewerApplication.page--;
+    });
+
+    document.getElementById('next').addEventListener('click', function() {
+      PDFViewerApplication.page++;
+    });
+
+    document.getElementById('zoomIn').addEventListener('click', function() {
+      PDFViewerApplication.zoomIn();
+    });
+
+    document.getElementById('zoomOut').addEventListener('click', function() {
+      PDFViewerApplication.zoomOut();
+    });
+
+    document.getElementById('pageNumber').addEventListener('click', function() {
+      this.select();
+    });
+
+    document.getElementById('pageNumber').addEventListener('change',
+        function() {
+      // Handle the user inputting a floating point number.
+      PDFViewerApplication.page = (this.value | 0);
+
+      if (this.value !== (this.value | 0).toString()) {
+        this.value = PDFViewerApplication.page;
+      }
+    });
+
+    container.addEventListener('pagesinit', function () {
+      // We can use pdfViewer now, e.g. let's change default scale.
+      pdfViewer.currentScaleValue = DEFAULT_SCALE_VALUE;
+    });
+
+    container.addEventListener('pagechange', function (evt) {
+      var page = evt.pageNumber;
+      if (evt.previousPageNumber !== page) {
+        document.getElementById('pageNumber').value = page;
+      }
+      var numPages = PDFViewerApplication.pagesCount;
+
+      document.getElementById('previous').disabled = (page <= 1);
+      document.getElementById('next').disabled = (page >= numPages);
+
```

Comment:
@yurydelendik Could you review/merge that PR first? Then I think this PR is also good to go. I have no further comments on the code in this PR.